internal/nearcache/repairing_task.go

```diff
@@ -0,0 +1,568 @@
+		cs:         cs,
+		is:         is,
+		invFactory: invFactory,
+		lg:         lg,
+	}
+	return df
+}
+
+func (df InvalidationMetaDataFetcher) Init(ctx context.Context, handler RepairingHandler) error {
+	handlers := map[string]RepairingHandler{
+		handler.Name(): handler,
+	}
+	return df.fetchMetadata(ctx, handlers)
+}
+
+func (df InvalidationMetaDataFetcher) fetchMetadata(ctx context.Context, handlers map[string]RepairingHandler) error {
+	// port of: com.hazelcast.internal.nearcache.impl.invalidation.InvalidationMetaDataFetcher#fetchMetadata
+	if len(handlers) == 0 {
+		return nil
+	}
+	// getDataStructureNames
+	names := []string{}
+	for _, h := range handlers {
+		names = append(names, h.Name())
+	}
+	invs, err := df.fetchMembersMetadataFor(ctx, names)
+	if err != nil {
+		return fmt.Errorf("InvalidationMetaDataFetcher.fetchMetadata: fetching members metadata: %w", err)
+	}
+	for _, inv := range invs {
+		if err := df.processMemberMetadata(ctx, inv, handlers); err != nil {
+			return fmt.Errorf("InvalidationMetaDataFetcher.fetchMetadata: processing metadata: %w", err)
+		}
+	}
+	return nil
+}
+
+func (df InvalidationMetaDataFetcher) fetchMembersMetadataFor(ctx context.Context, names []string) (map[types.UUID]invocation.Invocation, error) {
+	// port of: com.hazelcast.internal.nearcache.impl.invalidation.InvalidationMetaDataFetcher#fetchMembersMetadataFor
+	mems := filterDataMembers(df.cs.OrderedMembers())
+	if len(mems) == 0 {
+		return nil, nil
+	}
+	invs := make(map[types.UUID]invocation.Invocation, len(mems))
+	for _, mem := range mems {
+		inv, err := df.fetchMetaDataOf(ctx, mem, names)
+		if err != nil {
+			return nil, err
+		}
+		invs[mem.UUID] = inv
+	}
+	return invs, nil
+}
+
+func (df InvalidationMetaDataFetcher) fetchMetaDataOf(ctx context.Context, mem pubcluster.MemberInfo, names []string) (*cluster.MemberBoundInvocation, error) {
+	// port of: com.hazelcast.client.map.impl.nearcache.invalidation.ClientMapInvalidationMetaDataFetcher#fetchMetadataOf
+	msg := codec.EncodeMapFetchNearCacheInvalidationMetadataRequest(names, mem.UUID)
+	inv := df.invFactory.NewMemberBoundInvocation(msg, &mem, time.Now())
+	if err := df.is.SendRequest(ctx, inv); err != nil {
+		return nil, err
+	}
+	return inv, nil
+}
+
+func (df *InvalidationMetaDataFetcher) processMemberMetadata(ctx context.Context, inv invocation.Invocation, handlers map[string]RepairingHandler) error {
+	// port of: com.hazelcast.internal.nearcache.impl.invalidation.InvalidationMetaDataFetcher#processMemberMetadata
+	// extractMemberMetadata
+	res, err := inv.GetWithContext(ctx)
+	if err != nil {
+		return err
+	}
+	npsPairs, psPairs := codec.DecodeMapFetchNearCacheInvalidationMetadataResponse(res)
+	// repairUuids
+	// see: com.hazelcast.internal.nearcache.impl.invalidation.InvalidationMetaDataFetcher#repairUuids
+	for _, p := range psPairs {
+		k := p.Key.(int32)
+		v := p.Value.(types.UUID)
+		for _, handler := range handlers {
+			handler.CheckOrRepairUUID(k, v)
+		}
+	}
```

Comment:
We can make these separate methods as well.

Reply:
OK: https://github.com/hazelcast/hazelcast-go-client/pull/807/commits/0d49f98e13b8fc0d4843db1366d353fb6a96a174